include/common/logger.h

```diff
@@ -1 +1,69 @@
-class logger{};
\ No newline at end of file
+#pragma once
+
+#include <string>
+#include <fstream>
+#include <ctime>
+#include <chrono> 
+#include <assert.h>
+
+class logger {
+public:
+	enum log_level
+	{
+		ll_CRITICAL, 
+		ll_ERROR, 
+		ll_WARNING, 
+		ll_DEBUG, 
+		ll_INFO
+	};
+	logger(const std::string &file_name, log_level level) : m_full_buffer_size (100), log_level_needed (level)
+	{
+		m_fileStream.open(file_name, std::ofstream::out | std::ofstream::trunc);
+		assert(m_fileStream.is_open());
+	
+	}
+	~logger()
+	{
+		if (m_flushing_buffer_size.length() >= 1)
+		{
+			m_fileStream << m_flushing_buffer_size;
+		}
+		m_fileStream.close();
+	}
+	const std::string timenow()
+	{
+		auto timenow = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
```

Comment:
avoid using auto unless it can't be done in other way

Reply:
replaced by time_t utility class